src/main/webapp/forms/id.jsp

```diff
@@ -1,28 +1,49 @@
+<%@page import="de.uni_tuebingen.ub.nppm.db.EinzelbelegDB"%>
 <%@ include file="../configuration.jsp" %>
 
 <%
     String id = request.getParameter("ID");
     String title = request.getParameter("title");
+    String provenanceId = null;
+    String provenanceSrc = null;
     if (title.contains("gast_")) {
         title = title.substring(5);
     }
 
     out.println("ID: ");
     if (title.toLowerCase().equals("einzelbeleg")) {
         out.print("B");
+        provenanceId = EinzelbelegDB.getProvenanceId(id, "einzelbeleg");
+        provenanceSrc = EinzelbelegDB.getProvenanceSource(id, "einzelbeleg");
     } else if (title.toLowerCase().equals("person")) {
         out.print("P");
+        provenanceId = EinzelbelegDB.getProvenanceId(id, "person");
```

Comment:
The getProvenanceId method is implemented in AbstractBase. You should not try to use the EinzelbelegDB to call methods on entities for which it's not responsible. So please either use PersonDB, NamenKommentarDB... and so on here, or call AbstractBase directly. There are some other cases i already discussed with @rnaiser where we call AbstractBase (even though we shouldn't since the main idea was just having it as template for the subclasses) => There is a certain possibility that we might either rename AbstractBase to Base, or maybe add an intermediate GenericDB layer for these kinds of methods in the future.

Reply:
Yes, i have used now the specific db classes like PersonDB, NamenkommentarDB, MGHLemmaDB...